setup.py

```diff
@@ -591,5 +594,144 @@ def do_setup():
     )
 
 
+PROVIDERS_DEPENDENCIES = {
+    "amazon": [aws],
+    "apache.cassandra": [cassandra],
+    "apache.druid": [druid],
+    "apache.hdfs": [hdfs],
+    "apache.hive": [hive],
+    "apache.pig": [],
+    "apache.pinot": [pinot],
+    "apache.spark": [],
+    "apache.sqoop": [],
+    "celery": [celery],
+    "cloudant": [cloudant],
+    "cncf.kubernetes": [kubernetes],
+    "databricks": [databricks],
+    "datadog": [datadog],
+    "dingding": [],
+    "discord": [],
+    "docker": [docker],
+    "email": [],
+    "ftp": [],
+    "google.cloud": [gcp],
```

Comment:
For me it seems reasonable to release it as one package because all integration requires a common hook - CloudBaseHook. The division into various departments e.g. cloud, ads, marketing_platform, analytics can also change over time.  Currently, this is not a big problem, because we focused mainly on GCP, where the division into departments is clear, but other services do not have a clear division. For example: Google Ads is not related to Google Marketing Platform, Google Analytics is also not related to GMP.  Why? I don't know that, but for new people, understanding these structures will be problematic.

Reply:
This is up for discussion how we are going to release backport packages (see the devlist).

 I am also for releasing one big package for now - it will be far easier to manage. But then it will have a lot of dependencies. However keeping dependencies for each package in this dictionary might still make sense. To be discussed.